src/core/linalg/tests/4C_linalg_vector_test.np2.cpp

```diff
@@ -289,14 +293,25 @@ namespace
       my_elements = {1, 3, 5, 7, 9};
     Core::LinAlg::Map new_map(10, my_elements.size(), my_elements.data(), 0, comm_);
 
-    const Core::LinAlg::MultiVector<double>& b = a;
-    const Core::LinAlg::Vector<double>& c = b(0);
+    // Before replacement, all maps are the same.
+    EXPECT_TRUE(a.get_map().same_as(b.get_map()));
+    EXPECT_TRUE(a.get_map().same_as(c.get_map()));
 
-    // A change of the map is reflected to all views
+    EXPECT_EQ(&a.get_ref_of_epetra_vector(), &b.get_epetra_multi_vector());
+    // A change of the map invalidates views, so we need to be careful.
     a.replace_map(new_map);
+    EXPECT_NE(&a.get_ref_of_epetra_vector(), &b.get_epetra_multi_vector());
 
-    EXPECT_TRUE(a.get_map().same_as(b.get_map()));
-    EXPECT_TRUE(a.get_map().same_as(c.get_map()));
+    {
+      // This highlights a bug in Trilinos: the Epetra_Vector views into a MultiVector are only
```

Comment:
Do you know, if this is also the case for Tpetra? This should be reported to Trilinos, so that it can be fixed at its root.

Reply:
I had a quick look at Tpetra and I at least do not see the same problem right away -- although I have looked very little into this package and might simply miss it.